### PR TITLE
Ghost roles

### PR DESCRIPTION
**Pull request by Zumorica**
looking for some feedback on what I have so far

### CODE REVIEW

File: Content.Client/UserInterface/GhostRoleWindow.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace Content.Client.UserInterface
+{
+    public class GhostRoleWindow : SS14Window
+    {
+        protected override void Opened()
+        {
+            base.Opened();
+
+            EntitySystem.Get<GhostRoleSystem>()
```
Comment (RemieRichards) [inline Content.Client/UserInterface/GhostRoleWindow.cs:13]: very epic

Comment (Zumorica) [reply Content.Client/UserInterface/GhostRoleWindow.cs:13]: reee this is a draft I still gotta work on the UI itself

Comment (RemieRichards) [reply Content.Client/UserInterface/GhostRoleWindow.cs:13]: naughty naughty

File: Content.Server/GameObjects/Components/Observer/GhostRoleComponent.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.Serialization;
+
+namespace Content.Server.GameObjects.Components.Observer
+{
+    public abstract class GhostRoleComponent : Component
+    {
+        public string RoleName { get; set; }
+        public string RoleDescription { get; set; }
+        public bool Taken { get; set; }
```
Comment (ComicIronic) [inline Content.Server/GameObjects/Components/Observer/GhostRoleComponent.cs:13]: Does it make sense for these to be settable?

Comment (Zumorica) [reply Content.Server/GameObjects/Components/Observer/GhostRoleComponent.cs:13]: Not really, I could make the setters private without issue

**Comment by F77F** [comment]
> [x] The PR does not include any unnecessary .meta, .prefab or .unity (scene) changes

🤔

**Comment by RemieRichards** [comment]
@F77F it's just a meme

File: Content.Server/GameObjects/Components/Observer/GhostRoleMobSpawnerComponent.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            if(string.IsNullOrEmpty(Prototype))
+                throw new NullReferenceException("Prototype string cannot be null or empty!");
+
+            Taken = true;
+            var mob = _entityMan.SpawnEntity(Prototype, Owner.Transform.GridPosition);
+
+            mob.EnsureComponent<MindComponent>();
+            session.ContentData().Mind.TransferTo(mob);
+
+            Owner.Delete();
```
Comment (RemieRichards) [inline Content.Server/GameObjects/Components/Observer/GhostRoleMobSpawnerComponent.cs:46]: Maybe make this optional? ie: a bool field that's available in the yaml as `delete_on_spawn` or something

File: Content.Server/GameObjects/EntitySystems/GhostRoleSystem.cs [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+        public override void Initialize()
+        {
+            SubscribeNetworkEvent<GhostRoleUpdateRequestMessage>(GhostRolesUpdateRequest);
+            SubscribeNetworkEvent<GhostRoleTakeoverRequestMessage>(GhostRoleTakeoverRequest);
+        }
+
+        private uint GetNextRoleIdentifier()
+        {
+            return unchecked(_nextRoleIdentifier++);
```
Comment (RemieRichards) [inline Content.Server/GameObjects/EntitySystems/GhostRoleSystem.cs:32]: are there really gonna be that many ghost roles

**Comment by RemieRichards** [comment]
@Zumorica it looks like a good skeleton (spook!) for a ghost (spook!!) role system

**Comment by F77F** [comment]
No need for a thumbs down! I just didn't know why the anti Unity rhetoric was on github!

**Comment by Zumorica** [comment]
> No need for a thumbs down! I just didn't know why the anti Unity rhetoric was on github!

Dude, it's just a reaction emoji on a shitty github comment. Can you not disrupt the PR's comments anymore

**Comment by F77F** [comment]
Toxic!

**Comment by SweptWasTaken** [comment]
Closing until Zumorica feels like finishing it.

**Pull request closed without merging**